mgmake/generate.go

```diff
@@ -69,20 +73,39 @@ func GenerateMakefiles(mks ...Makefile) {
 		}
 		makefiles[toMakeTarget(namespace)] = makefile{Path: i.Path, DefaultTarget: defaultTarget}
 	}
+	ctx := logr.NewContext(context.Background(), mglogr.New("mage-tools-build"))
+	logr.FromContextOrDiscard(ctx).Info("building binary and generating Makefiles...")
+	GenMakefiles(ctx)
 }
 
-func GenMakefiles(ctx context.Context, exec string) error {
+func GenMakefiles(ctx context.Context) {
 	if len(makefiles) == 0 {
-		return fmt.Errorf("no makefiles to generate, see https://github.com/einride/mage-tools#readme for more info")
+		panic("no makefiles to generate, see https://github.com/einride/mage-tools#readme for more info")
 	}
-	executable = exec
-	targets, err := listTargets(ctx)
+	mageFilesCmd := mgtool.Command(ctx, "go", "list", "-e", "-f", `{{join .GoFiles "\n"}}`)
+	var b bytes.Buffer
+	mageFilesCmd.Stdout = &b
```

Comment:
Just an FYI that this can be replaced with `.Output()` which is run and output in a single command. Doesn't work when stdout is non-nil though. Any reason we're using this pattern instead? :thinking:

Reply:
Actually, i just changed this to `filepath.WalkDir`, was looking a bit to much at the mage implementation when i wrote it :)